scanner/updater/manual/manual.go

```diff
@@ -1,39 +1,252 @@
+	// DefaultURL Default URL to fetch the vulnerabilities JSON.
+	DefaultURL = "https://raw.githubusercontent.com/stackrox/stackrox/master/scanner/updater/manual/vulns.json"
+)
+
+var defaultClient = &http.Client{
+	Transport: proxy.RoundTripper(),
+	Timeout:   5 * time.Minute,
+}
+
+type updater struct {
+	c         *http.Client
+	updateURL *url.URL
+}
+
+// NewUpdater creates a new instance of the updater with default settings.
+func NewUpdater(c *http.Client, uri string) (*updater, error) {
+	var url *url.URL
+	var err error
+	if uri == "" {
+		uri = DefaultURL
+	}
+	url, err = url.Parse(uri)
 	if err != nil {
-		return res, err
+		return nil, err
 	}
-	return res, nil
+	return &updater{c: c, updateURL: url}, nil
 }
 
-type updater struct{}
+// Configure allows configuration of the updater, including changing the source URL.
+func (u *updater) Configure(ctx context.Context, f driver.ConfigUnmarshaler, c *http.Client) error {
+	ctx = zlog.ContextWithValues(ctx, "component", "updater/manual/manual.Configure")
+	var err error
+	if c == nil {
+		u.c = defaultClient
+	} else {
+		u.c = c
+	}
 
-// Name provides the name for the updater.
-func (u *updater) Name() string { return Name }
+	var cfg UpdaterConfig
+	if err := f(&cfg); err != nil {
+		return err
+	}
+	if cfg.URI != "" {
+		u.updateURL, err = url.Parse(cfg.URI)
+		if err != nil {
+			return err
+		}
+	}
+	zlog.Debug(ctx).Msg("loaded incoming config")
+	return nil
+}
+
+type UpdaterConfig struct {
+	Client *http.Client
+	URI    string
+}
 
-// Fetch returns nil values, as the updater does not fetch data.
-func (u *updater) Fetch(_ context.Context, _ driver.Fingerprint) (io.ReadCloser, driver.Fingerprint, error) {
-	return nil, "", nil
+func (u *updater) Name() string {
+	return Name
 }
 
-// Parse returns the manually added vulnerabilities.
-func (u *updater) Parse(_ context.Context, _ io.ReadCloser) ([]*claircore.Vulnerability, error) {
-	return u.vulns(), nil
+// Fetch fetching data from a configurable URI.
+func (u *updater) Fetch(ctx context.Context, fingerprint driver.Fingerprint) (io.ReadCloser, driver.Fingerprint, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "updater/manual/manual.Fetch")
+	resp, err := u.c.Get(u.updateURL.String())
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to fetch data: %w", err)
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			zlog.Error(ctx).Err(err).Msg("failed to close response body")
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, "", fmt.Errorf("bad response status: %s", resp.Status)
+	}
+	tempDir, err := os.MkdirTemp("", "scanner-manual-tmp")
+	if err != nil {
+		return nil, "", err
+	}
+	out, err := os.CreateTemp("", "manual")
+	if err != nil {
+		return nil, "", errors.Wrap(err, "creating scanner defs file")
+	}
+	zlog.Debug(ctx).
+		Str("filename", out.Name()).
+		Msg("opened temporary file for output")
+
+	_, err = io.Copy(out, resp.Body)
+	_ = os.RemoveAll(out.Name())
```

Comment:
Did this fail when you removed it before `io.Copy()`?

Reply:
I never tried to execute this before `io.Copy()`
Do we need to worry about the order?

Reply:
> Do we need to worry about the order?

The copy call doesn't need the name, does it? So, removing it before makes things clearer.